src/week2.hs

```diff
@@ -0,0 +1,202 @@
+{-# OPTIONS_GHC -Wall #-}
+--module week2 where
+
+-- Mastermind -----------------------------------------
+
+-- A peg can be one of six colors
+data Peg = Red | Green | Blue | Yellow | Orange | Purple
+         deriving (Show, Eq, Ord, Enum)
+
+-- A code is defined to simply be a list of Pegs
+type Code = [Peg]
+
+-- A move is constructed using a Code and two integers; the number of
+-- exact matches and the number of regular matches
+data Move = Move Code Int Int
+          deriving (Show, Eq)
+
+-- List containing all of the different Pegs
+colors :: [Peg]
+colors = [Red, Green, Blue, Yellow, Orange, Purple]
+
+-- Exercise 1 -----------------------------------------
+
+-- Get the number of exact matches between the actual code and the guess
+exactMatches :: Code -> Code -> Int
+
+exactMatches [] _ = 0
+exactMatches _ [] = 0
+exactMatches (x:xs) (x':xs') 
+    |x == x' = 1 + (exactMatches xs xs')
+    |otherwise = (exactMatches xs xs')
+
+-- or, using zipWith (Dmitry's suggestion)
+sumIfEqual :: Peg -> Peg -> Int
+sumIfEqual x y = if (x == y) then 1 else 0
+
+exactMatches1 :: Code -> Code -> Int
+exactMatches1 x y = sum(zipWith (sumIfEqual) x y)
```

Comment:
```suggestion
exactMatches1 x y = sum (zipWith (sumIfEqual) x y)
```
purely a stylistic manner, but I'd recommend including that space; the reasoning for me is that it's important to distinguish the role the parentheses play here---to tell the compiler to first compute the expression inside of it before applying `sum`, rather than what in traditional programming languages or in old school math is used to denote the function application itself